sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -73,8 +73,10 @@ case class Project(projectList: Seq[NamedExpression], child: LogicalPlan) extend
  * their output.
  *
  * @param generator the generator expression
- * @param join  when true, each output row is implicitly joined with the input tuple that produced
- *              it.
+ * @param unrequiredChildOutput each output row is implicitly joined with the relevant part from the
```

Comment:
Is it mandatory to use negative ones? `requiredChildOutput` seems more straightforward.

Reply:
Generally I agree, but in this case we will have to "fill" it in a few places when we initialise the object instead of starting it with Nil sequence. this way all the work is done entirely in the Optimizer.

Reply:
will try improve.